script.js

```diff
@@ -1,4 +1,709 @@
+      diceRollArray = rollAnyQtyDice(numberOfDice);
+      console.log("diceRollArray", diceRollArray);
+      // Store the dice roll in an unsequenced array (for output message use)
+      var unsortedDiceRollArray = storeRandomArray(diceRollArray);
+      // Get best combined number of the dice rolled
+      var bestNumber = autoCombine(resultMode, diceRollArray);
+      // Store best combined number in the array along with other players' best numbers
+      competingNumbers.push(Number(bestNumber));
+      // Extract all of current player's best numbers thus far into the individual player's personal array of best numbers
+      var playerCount1 = playerNo;
+      console.log("playerNo", playerNo);
+      console.log("playerCount1", playerCount1);
+      var indivArray = [];
+      while (playerCount1 < competingNumbers.length) {
+        indivArray.push(competingNumbers[playerCount1]);
+        console.log("playerCount1", playerCount1);
+        console.log(
+          "competingNumbers[playerCount1]",
+          competingNumbers[playerCount1]
+        );
+        console.log("indivArray", indivArray);
+        playerCount1 += Number(numberOfPlayers);
+      }
+      console.log("competingNumbers", competingNumbers);
+      console.log("indivArray", indivArray);
+      // If there is none stored yet (i.e. still in first round), use push to get new data in
+      if (competingArrays.length < numberOfPlayers) {
+        competingArrays.push(indivArray);
+      }
+      // If there is already data stored within (i.e. second round and beyond), replace old data
+      if (competingArrays.length >= numberOfPlayers) {
+        competingArrays[playerNo] = indivArray;
+      }
+      console.log("competingArrays", competingArrays);
+
+      // Collate the total of all the numbers in the individual player's personal array
+      var indivTotal = addTotal(indivArray);
+      // If there is none stored yet (i.e. still in first round), use push to get new data in
+      if (competingTotals.length < numberOfPlayers) {
+        competingTotals.push(indivTotal);
+      }
+      // If there is already data stored within (i.e. second round and beyond), replace old data
+      if (competingTotals.length >= numberOfPlayers) {
+        competingTotals[playerNo] = indivTotal;
+      }
+      console.log("competingTotals", competingTotals);
+
+      // Step 4: Output message (includes finding the winner if it is the last player of the round)
+      // If this is not the final player's turn, simply output the current player's dice roll outcome
+      if (playerNo == 0) {
+        playerNo += 1;
+        return `You rolled these numbers: ${unsortedDiceRollArray}. <br> <br> This is your best combined number from those digits above: ${bestNumber}. <br><br> It's the next player's turn now, please click submit to roll dice.`;
+      }
+
+      // If this is neither the first or last player's turn
+      if (playerNo != 0 && playerNo != numberOfPlayers - 1) {
```

Comment:
One way to prevent this complexity, would be to have a boolean variable as a global variable, something like 
`var winnerFound = true`
or
`var gameEnd = false`